adapters/oidc/js/src/main/resources/keycloak.d.ts

```diff
@@ -271,15 +279,14 @@ declare namespace Keycloak {
 	 * Conditional CompatPromise type in order to support
 	 * both legacy promises and native promises as return types.
 	 */
-	type PromiseType = KeycloakPromiseType | undefined;
-	type CompatPromise<TPromiseType extends PromiseType, TSuccess, TError> = 

```

Comment:
I tried to work with the types that were provided initially here but when creating a Keycloak instance I always got the native promise version even if the generic was never passed properly. I assume this was a bug which should now be fixed.

So just to clarify:

```
const instance = Keycloak(); // Typescript assumes this is a 'legacy' promise all methods return .success() and .error()
const instance = Keycloak<'native'>(); // Typescript assumes this is a native promise all methods return standard promises
```